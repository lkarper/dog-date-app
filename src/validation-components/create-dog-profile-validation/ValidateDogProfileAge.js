import React, { useEffect } from 'react';

const ValidateDogProfileAge = (props) => {

    const { ageYears, ageMonths, ageError, setAgeError } = props;

    useEffect(() => {
        if (!ageYears && !ageMonths) {
            setAgeError(`You must enter an age.`);
        } else {
            setAgeError('');
        }
    }, [ageYears, ageMonths, setAgeError]);

    if (ageError) {
        return <p id='age-validator'>{ageError}</p>;
    }

    if (ageYears && ageMonths) {
        return <p id='age-validator'>Age set to: {ageYears}{' '}{ageYears === 1 ? `year` : `years`}, {ageMonths}{' '}{ageMonths === 1 ? `month ` : `months `}old.</p>;
    }

    if (ageYears) {
        return <p id='age-validator'>Age set to: {ageYears}{' '}{ageYears === 1 ? `year ` : `years `}old.</p>;
    }

    return <p id='age-validator'>Age set to: {ageMonths}{' '}{ageMonths === 1 ? `month ` : `months `}old.</p>;

}

export default ValidateDogProfileAge;