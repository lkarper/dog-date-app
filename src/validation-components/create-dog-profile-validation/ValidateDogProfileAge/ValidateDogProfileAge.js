import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateDogProfileAge = (props) => {
    const { 
        ageYears, 
        ageMonths, 
        ageError, 
        setAgeError,
    } = props;

    useEffect(() => {
        if (!ageYears && !ageMonths) {
            setAgeError(`You must enter an age.`);
        } else {
            setAgeError('');
        }
    }, [ageYears, ageMonths, setAgeError]);

    if (ageError) {
        return (
            <p
                className='ValidateDogProfileAge__validator error' 
                id='age-validator'
            >
                {ageError}
            </p>
        );
    }

    if (ageYears && ageMonths) {
        return (
            <p 
                className='ValidateDogProfileAge__validator valid'
                id='age-validator'
            >
                Age set to: {ageYears}{' '}{ageYears === 1 ? `year` : `years`},{' '} 
                {ageMonths}{' '}{ageMonths === 1 ? `month ` : `months `}old.
            </p>
        );
    }

    if (ageYears) {
        return (
            <p 
                className='ValidateDogProfileAge__validator valid'
                id='age-validator'
            >
                Age set to: {ageYears}{' '}{ageYears === 1 ? `year ` : `years `}old.
            </p>
        );
    }

    return (
        <p 
            className='ValidateDogProfileAge__validator valid'
            id='age-validator'
        >
            Age set to: {ageMonths}{' '}{ageMonths === 1 ? `month ` : `months `}old.
        </p>
    );

}

ValidateDogProfileAge.defaultProps = {
    ageYears: '',
    ageMonths: '',
    ageError: 'You must enter an age.',
    setAgeError: () => {},
};

ValidateDogProfileAge.propTypes = {
    ageYears: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    ageMonths: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    ageError: PropTypes.string.isRequired,
    setAgeError: PropTypes.func.isRequired,
};

export default ValidateDogProfileAge;
