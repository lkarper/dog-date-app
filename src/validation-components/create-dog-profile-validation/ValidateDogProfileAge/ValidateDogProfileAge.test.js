import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateDogProfileAge from './ValidateDogProfileAge';

describe('ValidateDogProfileAge component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateDogProfileAge />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateDogProfileAge />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
