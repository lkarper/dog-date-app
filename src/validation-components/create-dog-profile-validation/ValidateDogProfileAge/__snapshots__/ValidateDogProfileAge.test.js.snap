// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateDogProfileAge component renders the UI as expected 1`] = `
<ValidateDogProfileAge
  ageError="You must enter an age."
  ageMonths=""
  ageYears=""
  setAgeError={[Function]}
>
  <p
    className="ValidateDogProfileAge__validator error"
    id="age-validator"
  >
    You must enter an age.
  </p>
</ValidateDogProfileAge>
`;
